projects/packages/publicize/src/class-publicize.php

```diff
@@ -138,7 +145,15 @@ public function disconnect( $service_name, $connection_id, $_blog_id = false, $_
 	 * @return true
 	 */
 	public function receive_updated_publicize_connections( $publicize_connections ) {
-		set_transient( self::JETPACK_SOCIAL_CONNECTIONS_TRANSIENT, $publicize_connections, 3600 * 4 );
+		$expiry = 3600 * 4;
+		if ( ! set_transient( self::JETPACK_SOCIAL_CONNECTIONS_TRANSIENT, $publicize_connections, $expiry ) ) {
+			// If the transient has beeen set in another request, the call to set_transient can fail. If so,
+			// we can delete the transient and try again.
+			delete_transient( self::JETPACK_SOCIAL_CONNECTIONS_TRANSIENT );
```

Comment:
Nitpick, feel free to ignore: Should this be `$this->clear_connections_transient();`?